### PR TITLE
Add a guide for g5/g6 instance to  efa-cheatsheet.md

### PR DESCRIPTION
**Pull request by KeitaW**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by perifaws — APPROVED** [review]

**Comment by KeitaW** [comment]
https://github.com/NVIDIA/nccl/issues/922#issuecomment-1642282439

**Comment by KeitaW** [comment]
`g6: AMD EPYC 7R13 Processor`

File: 1.architectures/efa-cheatsheet.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ versions of your libfabric.
...
 | `NCCL_BUFFSIZE=xxx`             | Recommend to leave it out to use the default.                                                                                                                                                                                                                                                                                                         |
 | `RDMAV_FORK_SAFE=1`             | Do not use. This is a RDMA-core environment variable. Prefer `FI_EFA_FORK_SAFE` (if it still makes sense for your Linux kernel version). The two looks the same, but actually behaves very differently, especially on newer kernels, where `RDMAV_FORK_SAFE=1` can break things.                                                                      |
+| `NCCL_SHM_USE_CUDA_MEMCPY=1`             | Setting this when you run NCCL on g6/g5. It gives x2 performance in comparison to default memcpy                                                                  |
```
Comment (perifaws) [inline 1.architectures/efa-cheatsheet.md:24]: Add in which context it provides better perf, for example the type of workload

**Review by perifaws — CHANGES_REQUESTED** [review]

File: 1.architectures/efa-cheatsheet.md (hunk 1 above)
Comment (KeitaW) [reply 1.architectures/efa-cheatsheet.md:24]: I believe that the type of workload does not matter. Whenever we have GPU-to-GPU communication on those instances, we probably want to test this variable.

**Pull request merged**